geomstats/visualization.py

```diff
@@ -342,28 +343,41 @@ def draw(self, ax, **kwargs):
 class PoincareHalfPlane:
     """Class used to plot points in the Poincare Half Plane."""
 
-    def __init__(self, points=None):
+    def __init__(self, points=None, point_type='half-plane'):
         self.points = []
+        self.point_type = point_type
         if points is not None:
             self.add_points(points)
 
     def add_points(self, points):
-        if not gs.all(H2.belongs(points)):
-            raise ValueError(
-                'Points do not belong to the hyperbolic space.')
-        points = self.convert_to_half_plane_coordinates(points)
+        if self.point_type == 'extrinsic':
+            if not gs.all(H2.belongs(points)):
+                raise ValueError(
+                    'Points do not belong to the hyperbolic space '
+                    '(extrinsic coordinates)')
+            points = self.convert_to_half_plane_coordinates(points)
+        elif self.point_type == 'half-plane':
+            if not gs.all(POINCARE_HALF_PLANE.belongs(points)):
+                raise ValueError(
+                    'Points do not belong to the hyperbolic space '
+                    '(Poincare half plane coordinates).')
         if not isinstance(points, list):
             points = list(points)
         self.points.extend(points)
 
-    @staticmethod
-    def set_ax(ax=None):
+    def set_ax(self, points, ax=None):
         if ax is None:
             ax = plt.subplot()
-        ax_s = AX_SCALE
+        if self.point_type == 'extrinsic':
+            points = self.convert_to_half_plane_coordinates(points)
+        xmin = points[gs.argmin(points[:, 0]), 0]
+        xmax = points[gs.argmax(points[:, 0]), 0]
+        ymax = points[gs.argmax(points[:, 1]), 1]
+        xmargin = 0.1 * (xmax - xmin)
+        ymargin = 0.1 * ymax
         plt.setp(ax,
-                 xlim=(-ax_s, ax_s),
-                 ylim=(0., ax_s),
+                 xlim=(xmin - xmargin, xmax + xmargin),
+                 ylim=(0., ymax + ymargin),
```

Comment:
Is this necessary to manually set xlim/ylim? It is the only thing preventing the exemple from running with tensorflow, and matplotlib does a good job at defining the limits automatically!

Reply:
Good point !